### DISCUSSION TITLE
Default direction when calculating 1D-PSD for topography

### DISCUSSION THREAD

**Discussion by sitangshugk95** [ContactEngineering/SurfaceTopography#348] [Q&A] [answered]
Which direction does the `power_spectrum_from_profile` [function](https://github.com/ContactEngineering/SurfaceTopography/blob/03338baa21ba6da6e6e313d52b247a0c397e6599/SurfaceTopography/Uniform/PowerSpectrum.py#L40) use by default, to calculate the 1D-PSD?
If it is averaged, is there an argument that allows us to choose if it is using the X vs the Y direction, to ensure that it is scanned along the scan direction for AFM scans?

**Comment by pastewka** [comment, marked as answer]
It's always x-direction. If this is not the scan direction, then there is likely something wrong with the reader, please let me know about this.

You can transpose the surface before computing the PSD to get the result in the other direction. It's `.transpose().power_spectrum_from_profile()`
